config/config.go

```diff
@@ -640,91 +644,150 @@ func getStakingTLSCert(v *viper.Viper) (tls.Certificate, error) {
+	ephemeralSignerEnabled := v.GetBool(StakingEphemeralSignerEnabledKey)
+	contentKeyIsSet := v.IsSet(StakingSignerKeyContentKey)
+	keyPathIsSet := v.IsSet(StakingSignerKeyPathKey)
+	rpcSignerURLIsSet := v.IsSet(StakingRPCSignerKey)
+
+	signingKeyPath := getExpandedArg(v, StakingSignerKeyPathKey)
+
+	bools := bag.Of(ephemeralSignerEnabled, contentKeyIsSet, keyPathIsSet, rpcSignerURLIsSet)
+	if bools.Count(true) > 1 {
+		return nil, nil, errInvalidSignerConfig
+	}
+
+	cleanup := func() error { return nil }
+
+	switch {
+	case ephemeralSignerEnabled:
+		signer, err := localsigner.New()
 		if err != nil {
-			return nil, fmt.Errorf("couldn't generate ephemeral signing key: %w", err)
+			return nil, nil, fmt.Errorf("couldn't generate ephemeral signing signer: %w", err)
 		}
-		return key, nil
-	}
 
-	if v.IsSet(StakingSignerKeyContentKey) {
+		return signer, cleanup, nil
+
+	case contentKeyIsSet:
 		signerKeyRawContent := v.GetString(StakingSignerKeyContentKey)
 		signerKeyContent, err := base64.StdEncoding.DecodeString(signerKeyRawContent)
 		if err != nil {
-			return nil, fmt.Errorf("unable to decode base64 content: %w", err)
+			return nil, nil, fmt.Errorf("unable to decode base64 content: %w", err)
 		}
-		key, err := localsigner.FromBytes(signerKeyContent)
+
+		signer, err := localsigner.FromBytes(signerKeyContent)
 		if err != nil {
-			return nil, fmt.Errorf("couldn't parse signing key: %w", err)
+			return nil, nil, fmt.Errorf("couldn't parse signing key: %w", err)
 		}
-		return key, nil
-	}
 
-	signingKeyPath := getExpandedArg(v, StakingSignerKeyPathKey)
-	_, err := os.Stat(signingKeyPath)
-	if !errors.Is(err, fs.ErrNotExist) {
-		signingKeyBytes, err := os.ReadFile(signingKeyPath)
+		return signer, cleanup, nil
+
+	case keyPathIsSet:
+		// If the key is set, but a user-file isn't provided, we don't create one.
+		// The siging key is only stored to the default file-location if it's created
+		// and saved by the current application run.
+		_, err := os.Stat(signingKeyPath)
+
+		if errors.Is(err, fs.ErrNotExist) {
+			return nil, nil, errMissingStakingSigningKeyFile
+		}
+
+		signer, err := createSignerFromFile(signingKeyPath)
 		if err != nil {
-			return nil, err
+			return nil, nil, fmt.Errorf("couldn't parse signing key: %w", err)
 		}
-		key, err := localsigner.FromBytes(signingKeyBytes)
+
+		return signer, cleanup, nil
+
+	case rpcSignerURLIsSet:
+		rpcSignerURL := v.GetString(StakingRPCSignerKey)
+
+		signer, cleanup, err := rpcsigner.NewClient(ctx, rpcSignerURL)
 		if err != nil {
-			return nil, fmt.Errorf("couldn't parse signing key: %w", err)
+			return nil, nil, fmt.Errorf("couldn't create rpc signer client: %w", err)
+		}
+
+		return signer, cleanup, nil
+
+	default:
+		_, err := os.Stat(signingKeyPath)
+
+		// if the file exists, try to load the key from the file
+		if !errors.Is(err, fs.ErrNotExist) {
+			signer, err := createSignerFromFile(signingKeyPath)
+			if err != nil {
+				return nil, nil, errors.Join(err, cleanup())
```

Comment:
I don't think cleanup is doing anything here.
```suggestion
				return nil, nil, err
```